File: web/src/domain/entities/project.ts
// 用户拥有的工作项目;name 是文件系统 slug(后端生成),title 是用户可编辑显示名。
export interface Project {
  id: string;
  name: string;
  title: string;
  // 容器内 cwd,iframe src 拼接用,不展示给用户。
  cwd: string;
  // ready / archived 等。
  status: string;
  updatedAt: string;
}

// 当前 active project 精简视图;null 表示还没建 / active 被清了。
export interface ActiveProject {
  id: string;
  name: string;
  title: string;
  cwd: string;
  // Sandbox-stamped CDN URL; "" until controller stamps it.
  previewUrl?: string;
  updatedAt: string;
}

File: internal/infra/data/model/project.go
package model

import "time"

type Project struct {
	ID                   string  `gorm:"column:id;primaryKey;type:uuid;default:gen_random_uuid()"`
	UserID               int64   `gorm:"column:user_id;not null;index"`
	Name                 string  `gorm:"column:name;not null;size:64;index"`
	Title                string  `gorm:"column:title;not null;default:''"`
	Status               string  `gorm:"column:status;not null;default:'creating';index"`
	CreatedFromMessageID *string `gorm:"column:created_from_message_id;type:uuid;index"`
	// Controller-stamped CDN URL; "" until reconcile completes.
	PreviewURL string     `gorm:"column:preview_url;not null;default:''"`
	CreatedAt  time.Time  `gorm:"column:created_at;autoCreateTime"`
	UpdatedAt  time.Time  `gorm:"column:updated_at;autoUpdateTime"`
	ArchivedAt *time.Time `gorm:"column:archived_at;index"`
}

func (Project) TableName() string { return "projects" }
